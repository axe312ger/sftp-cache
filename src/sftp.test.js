const { resolve } = require('path')

const { getRemoteFiles } = require('./sftp')

test('parses remote directory', async () => {
  const remoteCacheDir = '/home/cache/some/dir/'
  const dir = resolve(remoteCacheDir, 'dir-to-cache')

  const sftp = {
    readdir: (dirToRead, cb) => {
      if (dirToRead === dir) {
        return cb(false, [
          {
            filename: 'example-file-name.md',
            longname: 'file',
            attrs: { size: '123', mtime: '456' }
          },
          {
            filename: 'subdir',
            longname: 'directory',
            attrs: {}
          }
        ])
      }
      if (dirToRead === resolve(dir, 'subdir')) {
        return cb(false, [
          {
            filename: 'file-with-content.md',
            longname: 'file',
            attrs: { size: '123', mtime: '456' }
          }
        ])
      }

      return cb(new Error('oops'))
    }
  }

  const tree = await getRemoteFiles({ sftp, dir, remoteCacheDir })

  expect(tree).toMatchSnapshot()
})
