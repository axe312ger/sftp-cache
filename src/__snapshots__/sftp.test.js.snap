// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses remote directory 1`] = `
Array [
  Object {
    "name": "example-file-name.md",
    "path": "dir-to-cache/example-file-name.md",
    "stats": Object {
      "mtime": "456",
      "size": "123",
    },
  },
  Object {
    "name": "file-with-content.md",
    "path": "dir-to-cache/subdir/file-with-content.md",
    "stats": Object {
      "mtime": "456",
      "size": "123",
    },
  },
]
`;
